File: function.go
package p

import (
  "encoding/json"
  "fmt"
  "html"
  "net/http"

  "cloud.google.com/go/firestore"
)


func CreateCars(w http.ResponseWriter, r *http.Request) {

  var car struct {
    Plaque string `json:"plaque"`
    Color string `json:"color"`
    Price string `json:"price"`
    CarModel string `json:"carModel"`
    Brand string `json:"brand"`
  }

  ctx := context.Background()

	if err := json.NewDecoder(r.Body).Decode(&car); err != nil {
		fmt.Fprint(w, "Body inválido!")
		return
  }

  client, err := firestore.NewClient(ctx, "projectID")
  if err != nil {
    fmt.Fprint(w, "Erro ao conectar firestore.")
  }

  wr, err := ny.Create(ctx, car)

  if err != nil {
    fmt.Fprint(w, "Erro ao salvar o carro.")
  }

  fmt.Println(wr)

	fmt.Fprint(w, html.EscapeString({car.Plaque, car.Color, car.Price, car.CarModel, car.Brand}))
}

File: function.js
const Firestore = require('@google-cloud/firestore');

exports.createCar = async (req, res) => {
  const {
    plaque,
    color,
    price,
    brand,
    carModel,
  } = req.body;

  const firestore = new Firestore({
    projectId: 'PROJECT_ID',
    keyFilename: './keyfile.json',
  });

  const docRef = firestore.collection('americanCar').doc('car');

  await docRef.set({
    plaque,
    color,
    price,
    brand,
    carModel,
  })

  res.status(200).send('Car created!');
};

exports.findCarByPlaque = async (req, res) => {
  const { plaque } = req.query;

  console.log('PARAMS -> ', req.query);
  console.log('PLAQUE -> ', plaque);

  const firestore = new Firestore({
    projectId: 'PROJECT_ID',
    keyFilename: './keyfile.json',
  });

  const carsRef = firestore.collection('americanCar');

  const car = await carsRef.where('plaque', '==', plaque).get();

  if (car.empty) {
    console.log('No car found for this plaque!');

    return res.status(404).send('No car found for this plaque!');
  }

  console.log('CAR -> ', car.data());

  res.status(200).send(car.data());
};
